modules/ROOT/pages/dataweave-cookbook-set-reader-writer-props.adoc

```diff
@@ -0,0 +1,192 @@
+
+The following examples pass reader properties through the `outputMimeType` field.
+
+* xref:dataweave-formats-csv.adoc#example2[CSV example] that uses `streaming=true; header=true`
+* xref:dataweave-formats-json.adoc#example3[JSON example} that uses `streaming=true`
+
+[[ex04]]
+== Use a Mule Variable as a Configuration Value
+
+The following Mule flow transforms comma-separated CSV data into pipe-separated (`|`) CSV data. Instead of setting a literal value in the script, the example selects the pipe value from a xref:about-mule-variables.adoc[Mule variable], which is part of a Mule event that travels through data-processing components in the flow.
+
+The Scheduler component (`<scheduler/>`) in the example generates a Mule event each time it triggers the flow.
+
+The Set Variable component (`<set-variable/>`) creates a Mule variable named `delimiter` with the value `'|'`. In Studio, the value is created as an *fx* expression (`value="#['|']"`), rather than a simple string (`value="'|'"`).
+
+The Transform Message component (`<ee:transform-message/>`) contains a DataWeave script that reads comma-separated CSV input and uses the Mule variable to write pipe-separated CSV output.
+
+The Logger component (`<logger/>`), which is set to `payload`, writes the pipe-separated output as INFO in its `LoggerMessageProcessor` message: `macaroni | rigatoni | ravioli | spaghetti`.
+
+.Mule Flow:
+[source,xml,linenums]
+----
+<flow name="ex-use-mule-var-as-prop-config-val" >
+  <scheduler doc:name="Scheduler" >
+    <scheduling-strategy >
+      <fixed-frequency frequency="45" timeUnit="SECONDS"/>
+    </scheduling-strategy>
+  </scheduler>
+  <set-variable value="#['|']" doc:name="Set Variable" variableName="delimiter" />
+  <ee:transform doc:name="Transform Message" >
+    <ee:message>
+      <ee:set-payload><![CDATA[%dw 2.0
+var myVar = read("macaroni , rigatoni , ravioli , spaghetti", "application/csv" , {"header":false, separator:','})
+output application/csv with binary
+---
+write(myVar, "application/csv", {"header":false, "separator":vars.delimiter})]]></ee:set-payload>
+    </ee:message>
+  </ee:transform>
+  <logger level="INFO" doc:name="Logger" message="#[payload]"/>
+</flow>
+----
+
+To provide comma-separated input for the example, the script in the Transform Message component uses a xref:dw-core-functions-read.adoc[read] function as a value to a DataWeave variable, `myVar`:
+
+`var myVar = read("macaroni , rigatoni , ravioli , spaghetti", "application/csv" , {"header":false, separator:','})`
+
+To write pipe-separated the output, the DataWeave xref:dw-core-functions-write.adoc[write] function in the body of the script changes the comma-separated input to pipe-separated output. To select the pipe value from the Mule variable (`delimiter`), the function passes the writer property configuration `"separator":vars.delimiter` as an argument:
+
+`var myVar = write(myVar, "application/csv", {"header":false, "separator":vars.delimiter})`
```

Comment:
This does not match the script. Should be simply `write(myVar, "application/csv", {"header":false, "separator":vars.delimiter})`

Reply:
oops! good catch! :)